packages/engine/src/scene/systems/EnvironmentSystem.tsx

```diff
@@ -71,21 +71,21 @@ const EnvMapReactor = () => {
   const materialComponentEntities = useChildrenWithComponents(entity, [MaterialStateComponent])
   return (
     <>
-      {materialComponentEntities.map((materialComponentEntity) => {
+      {materialComponentEntities.map((materialComponentEntity, index) => {
         switch (envMapComponent) {
           case 'Skybox':
-            return <EnvMapSkyboxReactor entity={materialComponentEntity} rootEntity={entity} />
+            return <EnvMapSkyboxReactor key={materialComponentEntity + '-' + index} entity={materialComponentEntity} rootEntity={entity} />
```

Comment:
should probably key with the type as well as index, to ensure they are treated as different children but still memoize correctly

Reply:
Yeah

Reply:
added and fixed the error